src/jaybenne/sourcing.cpp

```diff
@@ -87,12 +101,44 @@ TaskStatus SourcePhotons(T *md, const Real t_start, const Real dt) {
               [[maybe_unused]] const Real &sbd = sb;
               [[maybe_unused]] const Real &vvd = vv;
               [[maybe_unused]] const Real &dtd = dt;
-              [[maybe_unused]] auto &opac = opacity;
+              [[maybe_unused]] auto opac = opacity;
+              [[maybe_unused]] auto mopac = mopacity;
+              [[maybe_unused]] const auto numind = numin;
+              [[maybe_unused]] const auto numaxd = numax;
+              [[maybe_unused]] const auto n_nubinsd = n_nubins;
               Real erad = 0.0;
               if constexpr (ST == SourceType::thermal) {
                 erad = (4.0 * sbd / vvd) * std::pow(temp, 4.0) * dv;
               } else if constexpr (ST == SourceType::emission) {
-                const Real emis = opac.Emissivity(rho, temp);
+                Real emis = -1.;
+                if constexpr (FT == FrequencyType::gray) {
+                  emis = mopac.Emissivity(rho, temp);
+                } else if constexpr (FT == FrequencyType::multigroup) {
+                  // Construct emission CDF
+                  const Real dlnu = (std::log(numaxd) - std::log(numind)) / n_nubinsd;
+                  Real nu = std::exp(std::log(numind) + 0.5 * dlnu);
+                  Real dnu = nu * dlnu;
+                  vmesh(b, fj::emission_cdf(0), k, j, i) =
+                      opac.EmissivityPerNu(rho, temp,
+                                           std::exp(std::log(numind) + 0.5 * dlnu)) *
+                      dnu;
+                  for (int n = 1; n < n_nubinsd; n++) {
+                    nu = std::exp(std::log(numind) + (n + 0.5) * dlnu);
+                    dnu = dlnu * nu;
```

Comment:
Same question here - should note the previous suggestion is only exactly correct if `nu` is the lower-bound, but it looks like the logarithmic midpoint, rather.